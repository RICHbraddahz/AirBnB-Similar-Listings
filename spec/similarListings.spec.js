import React from 'react';
import { shallow, mount, render } from 'enzyme'; 
import { shallowToJson } from 'enzyme-to-json';

import SimilarListings from '../client/src/similarListings';
import Listing from '../client/src/listing';

describe('Similar Listings', () => {
  const listing = {
    "id":199,
    "url":"mockUrl",
    "title":"nostrum laboriosam in rerum officiis",
    "type":"Shared Room",
    "numBeds":2,
    "price":810,
    "numRatings":965,
    "avgStars":0.5,
    "thumbnailImage":"mockImageUrl"
  };
  const listings = new Array(4).fill(listing);
  const shallowWrapper = shallow(<SimilarListings listings={listings} />);

  test('renders correctly', () => {
    expect(shallowToJson(shallowWrapper)).toMatchSnapshot();
  })

  test('renders three Listing components', () => {
    expect(shallowWrapper.find(Listing).length).toBe(3);
  }); 
});