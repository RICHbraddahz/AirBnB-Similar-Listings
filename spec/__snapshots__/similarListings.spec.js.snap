// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Similar Listings renders correctly 1`] = `
<div
  className="similar-listings"
>
  <h1>
    Similar Listings
  </h1>
  <div
    className="similar-listings-ribbon"
  >
    <Listing
      additionalDetails={
        Object {
          "avgStars": 0.5,
          "numBeds": 2,
          "numRatings": 965,
          "price": 810,
          "thumbnailImage": "mockImageUrl",
          "title": "nostrum laboriosam in rerum officiis",
          "type": "Shared Room",
        }
      }
      id={199}
      key="199"
      thumbnailImage="mockImageUrl"
      url="mockUrl"
    />
    <Listing
      additionalDetails={
        Object {
          "avgStars": 0.5,
          "numBeds": 2,
          "numRatings": 965,
          "price": 810,
          "thumbnailImage": "mockImageUrl",
          "title": "nostrum laboriosam in rerum officiis",
          "type": "Shared Room",
        }
      }
      id={199}
      key="199"
      thumbnailImage="mockImageUrl"
      url="mockUrl"
    />
    <Listing
      additionalDetails={
        Object {
          "avgStars": 0.5,
          "numBeds": 2,
          "numRatings": 965,
          "price": 810,
          "thumbnailImage": "mockImageUrl",
          "title": "nostrum laboriosam in rerum officiis",
          "type": "Shared Room",
        }
      }
      id={199}
      key="199"
      thumbnailImage="mockImageUrl"
      url="mockUrl"
    />
  </div>
</div>
`;
